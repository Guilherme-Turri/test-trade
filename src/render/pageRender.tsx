import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};