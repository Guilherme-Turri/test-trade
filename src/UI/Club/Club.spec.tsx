import { screen } from '@testing-library/react'
import { Club } from '.'
import { componentRender } from '../../render/componentRender'

describe('</Club>', () => {
  it('should render on screen', () => {
    const { container } = componentRender(<Club readyToChose={true} />)
    expect(screen.getByTestId('Club')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })
  it('should render on screen with grey color', () => {
    componentRender(<Club readyToChose={false} />)
    expect(screen.getByTestId('Club')).toBeInTheDocument();
    expect(screen.getByTestId('Club')).toHaveStyleRule('color', 'grey');
  })
})