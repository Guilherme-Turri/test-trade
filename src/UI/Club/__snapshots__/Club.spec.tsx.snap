// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</Club> should render on screen 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  color: #00c555;
  width: 100%;
  height: 100%;
}

<div>
  <div
    class="c0"
    data-testid="Club"
  >
    <svg
      aria-hidden="true"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m6 13.5 4 2.5 4-2.5V5H6v8.5zM4.5 10a2 2 0 1 0-4 0 2 2 0 0 0 4 0zm13-2a2 2 0 1 0 0 4 2 2 0 0 0 0-4zM4.485 6.199A6.71 6.71 0 0 1 10 3.3a6.715 6.715 0 0 1 5.456 2.823 1.4 1.4 0 0 0 2.281-1.624A9.518 9.518 0 0 0 10 .5a9.506 9.506 0 0 0-7.817 4.107 1.402 1.402 0 0 0 .355 1.948 1.401 1.401 0 0 0 1.947-.356zm10.971 7.678A6.713 6.713 0 0 1 10 16.7a6.71 6.71 0 0 1-5.515-2.899 1.4 1.4 0 0 0-2.302 1.592A9.506 9.506 0 0 0 10 19.5a9.518 9.518 0 0 0 7.737-3.999 1.401 1.401 0 0 0-2.281-1.624z"
      />
    </svg>
  </div>
</div>
`;
